src/target/riscv/riscv.c

```diff
@@ -1545,8 +1555,14 @@ static int riscv_examine(struct target *target)
 	/* Don't need to select dbus, since the first thing we do is read dtmcontrol. */
 
 	RISCV_INFO(info);
-	uint32_t dtmcontrol = dtmcontrol_scan(target, 0);
+	uint32_t dtmcontrol;
+	if (dtmcontrol_scan(target, 0, &dtmcontrol) != ERROR_OK || dtmcontrol == 0) {
+		LOG_TARGET_ERROR(target, "Could not read dtmcontrol. Check JTAG connectivity/board power.");
+		return ERROR_FAIL;
+	}
 	LOG_TARGET_DEBUG(target, "dtmcontrol=0x%x", dtmcontrol);
+	// FIXME: if examine fails we may need to restore the original dtm_version
```

Comment:
That makes sense. Can you add that change? Shouldn't be much more complicated.

Reply:
will try... though I'm not sure that we need to fix it in this patch. `examine` has it's share of issues:

for example, it halts the target, but if something goes wrong - the target is not resumed. The fix for such situations may be complicated. But having said that - I'll try to address at least `dtm_version` stuff

Reply:
addressed..

Apparently my whining about examine implementation is not relevant to `riscv.c`